# location_matching_20240409

Working with files from:
`googledrive/VECTOR ATLAS/2024-03-14 - VA old but updata data for cleaning`

![](google_drive.png)

Matching locations missing in: `final_species.csv`,

from: `OLD MAP database dump/20210726_vector_extraction (1).csv`

Packages

``` r
library(dplyr)
```


    Attaching package: 'dplyr'

    The following objects are masked from 'package:stats':

        filter, lag

    The following objects are masked from 'package:base':

        intersect, setdiff, setequal, union

``` r
library(tidyr)
library(readr)
library(countrycode)
```

Read in data files and look at them:

##### new data base

``` r
new_db <- read_csv(
  file = "2024-03-14 - VA old but updated data for cleaning/final_species.csv",
  guess_max = 30000
)
```

    New names:
    Rows: 22150 Columns: 306
    ── Column specification
    ──────────────────────────────────────────────────────── Delimiter: "," chr
    (147): MAP_DATA, MAP_NOTES, citation..DOI., author, article.title, journ... dbl
    (115): ...1, X.1, X, bio_data, occ_data, source_ID, publication.year, la... lgl
    (44): common.occupation_3, outdoor.activities.at.night, hours.spent.awa...
    ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
    Specify the column types or set `show_col_types = FALSE` to quiet this message.
    • `` -> `...1`

``` r
glimpse(new_db)
```

    Rows: 22,150
    Columns: 306
    $ ...1                                                  <dbl> 18121, 5, 60, 62…
    $ X.1                                                   <dbl> 29407, 3339, 925…
    $ X                                                     <dbl> 29407, 3339, 925…
    $ bio_data                                              <dbl> 0, 0, 0, 0, 0, 0…
    $ occ_data                                              <dbl> 1, 1, 1, 1, 1, 1…
    $ MAP_DATA                                              <chr> "NO", NA, NA, NA…
    $ MAP_NOTES                                             <chr> NA, "Authors men…
    $ source_ID                                             <dbl> 1050, 1462, 3248…
    $ citation..DOI.                                        <chr> NA, "10.1186_147…
    $ author                                                <chr> "Bockarie", "Cua…
    $ article.title                                         <chr> "Malaria in a ru…
    $ journal.title                                         <chr> "Ann Trop Med Pa…
    $ publication.year                                      <dbl> 1994, 2006, 2008…
    $ adult.data                                            <chr> "yes", "yes", "y…
    $ larval.site.data                                      <chr> NA, NA, NA, NA, …
    $ IR.data                                               <chr> "none", "genotyp…
    $ study_sampling.design                                 <chr> "na", "na", "na"…
    $ contact.authors                                       <chr> NA, NA, NA, NA, …
    $ source.notes                                          <chr> NA, NA, NA, NA, …
    $ country                                               <chr> "Sierra Leone", …
    $ site                                                  <chr> "Nengbema,Nyande…
    $ latitude_1                                            <dbl> 8.1340, -8.9833,…
    $ longitude_1                                           <dbl> -11.6510, 13.250…
    $ latitude_2                                            <dbl> 8.166, NA, NA, N…
    $ longitude_2                                           <dbl> -11.484, NA, NA,…
    $ latitude_3                                            <dbl> 8.2, NA, NA, NA,…
    $ longitude_3                                           <dbl> -11.468, NA, NA,…
    $ latitude_4                                            <dbl> 8.2, NA, NA, NA,…
    $ longitude_4                                           <dbl> -11.501, NA, NA,…
    $ latitude_5                                            <dbl> 8.117, NA, NA, N…
    $ longitude_5                                           <dbl> -11.618, NA, NA,…
    $ latitude_6                                            <dbl> NA, NA, NA, NA, …
    $ longitude_6                                           <dbl> NA, NA, NA, NA, …
    $ latitude_7                                            <dbl> NA, NA, NA, NA, …
    $ longitude_7                                           <dbl> NA, NA, NA, NA, …
    $ latitude_8                                            <dbl> NA, NA, NA, NA, …
    $ longitude_8                                           <dbl> NA, NA, NA, NA, …
    $ confidence.in.georef                                  <chr> NA, NA, NA, NA, …
    $ GOOD_GUESS                                            <chr> "no", "no", "no"…
    $ BAD_GUESS                                             <chr> "no", "no", "no"…
    $ area.type                                             <chr> "wide area", "la…
    $ georef.source                                         <chr> "*encarta", "com…
    $ admin.level_1                                         <chr> "not specified",…
    $ admin.level_2                                         <chr> "Bo district", N…
    $ site.notes                                            <chr> "Results present…
    $ insecticide.control                                   <chr> "n.a.", "n.a.", …
    $ control.type                                          <chr> NA, NA, NA, NA, …
    $ ITN.use                                               <chr> NA, NA, NA, NA, …
    $ control.notes                                         <chr> NA, NA, NA, NA, …
    $ sampling.method_1                                     <chr> "HLC - in", "Lar…
    $ n_1                                                   <dbl> 0, 1, 1, 1, 1, 0…
    $ sampling.method_2                                     <chr> "CDC-LT", NA, NA…
    $ n_2                                                   <dbl> 1, NA, NA, NA, N…
    $ sampling.method_3                                     <chr> "WinExit", NA, N…
    $ n_3                                                   <dbl> 0, NA, NA, NA, N…
    $ sampling.method_4                                     <chr> NA, NA, NA, NA, …
    $ n_4                                                   <dbl> NA, NA, NA, NA, …
    $ n_tot                                                 <dbl> 1, NA, 1, 1, 1, …
    $ sampling.method.notes                                 <chr> "Human bait trap…
    $ binary.presence                                       <chr> "yes", "yes", "y…
    $ binary.absence                                        <chr> "no", "no", "no"…
    $ abundance.data.in.a.graph                             <chr> "no", "no", "no"…
    $ month_st                                              <dbl> 1, 4, 1, 4, 6, 6…
    $ month_end                                             <dbl> 4, 4, 1, 4, 7, 9…
    $ year_st                                               <dbl> 1990, 2001, 2004…
    $ year_end                                              <dbl> 1991, 2001, 2004…
    $ season..given.                                        <chr> "cross", NA, NA,…
    $ season..calc.                                         <chr> NA, NA, NA, NA, …
    $ rainfall_time                                         <chr> "2500 mm_year", …
    $ season.notes                                          <chr> NA, NA, NA, NA, …
    $ species                                               <chr> "abscurus", "ara…
    $ SPECIES1                                              <chr> "abscurus", "GAM…
    $ SPECIES2                                              <chr> "n.a.", "arabien…
    $ species.notes                                         <chr> NA, NA, NA, NA, …
    $ id_1                                                  <chr> "unknown", "gene…
    $ id_2                                                  <chr> NA, "genetic_mol…
    $ roof                                                  <chr> NA, NA, NA, NA, …
    $ walls                                                 <chr> NA, NA, NA, NA, …
    $ house.screening                                       <chr> NA, NA, NA, NA, …
    $ open.eaves                                            <chr> NA, NA, NA, NA, …
    $ cooking                                               <chr> NA, NA, NA, NA, …
    $ housing.notes                                         <chr> NA, NA, NA, NA, …
    $ common.occupation_1                                   <chr> NA, NA, NA, NA, …
    $ common.occupation_2                                   <chr> NA, NA, NA, NA, …
    $ common.occupation_3                                   <lgl> NA, NA, NA, NA, …
    $ outdoor.activities.at.night                           <lgl> NA, NA, NA, NA, …
    $ sleeping.outdoors                                     <chr> NA, NA, NA, NA, …
    $ outdoor.timings_hours                                 <chr> NA, NA, NA, NA, …
    $ outdoor.activities.notes                              <chr> NA, NA, NA, NA, …
    $ average.bedtime                                       <chr> NA, NA, NA, NA, …
    $ average.wake.time                                     <chr> NA, NA, NA, NA, …
    $ time.people.leave.home..in.morning                    <chr> NA, NA, NA, NA, …
    $ hours.spent.away.from.home.per.day                    <lgl> NA, NA, NA, NA, …
    $ seasonal.labour                                       <lgl> NA, NA, NA, NA, …
    $ community.notes                                       <chr> NA, NA, NA, NA, …
    $ forest                                                <chr> "yes", NA, NA, N…
    $ farming                                               <chr> NA, NA, NA, NA, …
    $ farming.notes                                         <chr> NA, NA, NA, NA, …
    $ livestock_1                                           <chr> "goats", NA, NA,…
    $ livestock_2                                           <chr> "pigs", NA, NA, …
    $ livestock_3                                           <chr> "dogs", NA, NA, …
    $ livestock_4                                           <chr> "other", NA, NA,…
    $ livestock.notes                                       <chr> "Sheep were also…
    $ local.plants                                          <chr> NA, NA, NA, NA, …
    $ environment.notes                                     <chr> NA, NA, NA, NA, …
    $ sampling..biology._1                                  <chr> NA, NA, NA, NA, …
    $ sampling..biology._2                                  <chr> NA, NA, NA, NA, …
    $ sampling..biology._3                                  <chr> NA, NA, NA, NA, …
    $ sampling..biology._n                                  <chr> NA, NA, NA, NA, …
    $ parity..n.                                            <dbl> NA, NA, NA, NA, …
    $ parity..total.                                        <dbl> NA, NA, NA, NA, …
    $ parity..percent.                                      <dbl> NA, NA, NA, NA, …
    $ daily.survival.rate..percent.                         <dbl> NA, NA, NA, NA, …
    $ fecundity..mean.batch.size.                           <dbl> NA, NA, NA, NA, …
    $ gonotrophic.cycle..days.                              <dbl> NA, NA, NA, NA, …
    $ biology.notes                                         <chr> NA, NA, NA, NA, …
    $ sampling..infection._1                                <chr> NA, NA, NA, NA, …
    $ sampling..infection._2                                <chr> NA, NA, NA, NA, …
    $ sampling..infection._3                                <chr> NA, NA, NA, NA, …
    $ sampling..infection._n                                <chr> NA, NA, NA, NA, …
    $ SR.by.dissection..n.                                  <dbl> NA, NA, NA, NA, …
    $ SR.by.dissection..total.                              <dbl> NA, NA, NA, NA, …
    $ SR.by.dissection..percent.                            <dbl> NA, NA, NA, NA, …
    $ SR.by.CSP..n_pool.                                    <chr> NA, NA, NA, NA, …
    $ SR.by.CSP..total_pool.                                <chr> NA, NA, NA, NA, …
    $ SR.by.CSP.percent.                                    <dbl> NA, NA, NA, NA, …
    $ SR.by.Pf..n.                                          <dbl> NA, NA, NA, NA, …
    $ SR.by.Pf..total.                                      <dbl> NA, NA, NA, NA, …
    $ SR.by.P..falciparum                                   <dbl> NA, NA, NA, NA, …
    $ SR.by.Pv..n.                                          <lgl> NA, NA, NA, NA, …
    $ SR.by.Pv..total.                                      <lgl> NA, NA, NA, NA, …
    $ SR.by.P..vivax                                        <lgl> NA, NA, NA, NA, …
    $ oocyst..n.                                            <dbl> NA, NA, NA, NA, …
    $ oocyst..total.                                        <dbl> NA, NA, NA, NA, …
    $ oocyst.rate..percent.                                 <dbl> NA, NA, NA, NA, …
    $ EIR                                                   <dbl> NA, NA, NA, NA, …
    $ EIR..period.                                          <chr> NA, NA, NA, NA, …
    $ Ext..incubation.period..days.                         <dbl> NA, NA, NA, NA, …
    $ infection.notes                                       <chr> NA, NA, NA, NA, …
    $ HBR.sampling..indoor.                                 <chr> NA, NA, NA, NA, …
    $ indoor.HBR                                            <dbl> NA, NA, NA, NA, …
    $ HBR.sampling..outdoor.                                <chr> NA, NA, NA, NA, …
    $ outdoor.HBR                                           <dbl> NA, NA, NA, NA, …
    $ HBR.sampling..combined._1                             <chr> NA, NA, NA, NA, …
    $ HBR.sampling..combined._2                             <chr> NA, NA, NA, NA, …
    $ HBR.sampling..combined._3                             <chr> NA, NA, NA, NA, …
    $ HBR.sampling..combined._n                             <lgl> NA, NA, NA, NA, …
    $ combined.HBR                                          <dbl> NA, NA, NA, NA, …
    $ HBR..unit.                                            <chr> NA, NA, NA, NA, …
    $ ABR.sampling_1                                        <lgl> NA, NA, NA, NA, …
    $ ABR.sampling_2                                        <lgl> NA, NA, NA, NA, …
    $ ABR.sampling_3                                        <lgl> NA, NA, NA, NA, …
    $ ABR.sampling_n                                        <lgl> NA, NA, NA, NA, …
    $ ABR                                                   <lgl> NA, NA, NA, NA, …
    $ ABR.unit                                              <lgl> NA, NA, NA, NA, …
    $ biting.rate.notes                                     <chr> NA, NA, NA, NA, …
    $ host.sampling..indoor.                                <chr> NA, NA, NA, NA, …
    $ indoor.host..n.                                       <dbl> NA, NA, NA, NA, …
    $ indoor.host..total.                                   <dbl> NA, NA, NA, NA, …
    $ indoor.host.percent                                   <dbl> NA, NA, NA, NA, …
    $ host.sampling..outdoor.                               <chr> NA, NA, NA, NA, …
    $ outdoor.host..n.                                      <dbl> NA, NA, NA, NA, …
    $ outdoor.host..total.                                  <dbl> NA, NA, NA, NA, …
    $ outdoor.host..percent.                                <dbl> NA, NA, NA, NA, …
    $ host.sampling..combined._1                            <chr> NA, NA, NA, NA, …
    $ host.sampling..combined._2                            <chr> NA, NA, NA, NA, …
    $ host.sampling..combined._3                            <chr> NA, NA, NA, NA, …
    $ host.sampling..combined._n                            <chr> NA, NA, NA, NA, …
    $ combined.host..n.                                     <dbl> NA, NA, NA, NA, …
    $ combined.host..total.                                 <dbl> NA, NA, NA, NA, …
    $ combined.host                                         <dbl> NA, NA, NA, NA, …
    $ host..unit.                                           <chr> NA, NA, NA, NA, …
    $ host.sampling..other._1                               <chr> NA, NA, NA, NA, …
    $ host.sampling..other._2                               <lgl> NA, NA, NA, NA, …
    $ host.sampling..other._3                               <lgl> NA, NA, NA, NA, …
    $ host.sampling..other._n                               <chr> NA, NA, NA, NA, …
    $ other.host..n.                                        <chr> NA, NA, NA, NA, …
    $ other.host..total.                                    <dbl> NA, NA, NA, NA, …
    $ host..other.                                          <dbl> NA, NA, NA, NA, …
    $ host..other..unit                                     <chr> NA, NA, NA, NA, …
    $ host.notes                                            <chr> NA, NA, NA, NA, …
    $ biting....No..of.sampling.nights..indoors.            <dbl> NA, NA, NA, NA, …
    $ biting.sampling..indoor.                              <chr> NA, NA, NA, NA, …
    $ indoor.biting..n.                                     <dbl> NA, NA, NA, NA, …
    $ indoor.biting..total.                                 <dbl> NA, NA, NA, NA, …
    $ indoor.biting.data                                    <dbl> NA, NA, NA, NA, …
    $ biting....No..of.sampling.nights..outdoors.           <dbl> NA, NA, NA, NA, …
    $ biting.sampling..outdoor.                             <chr> NA, NA, NA, NA, …
    $ outdoor.biting..n.                                    <dbl> NA, NA, NA, NA, …
    $ outdoor.biting..total.                                <dbl> NA, NA, NA, NA, …
    $ outdoor.biting.data                                   <dbl> NA, NA, NA, NA, …
    $ indoor_outdoor.biting..unit.                          <chr> NA, NA, NA, NA, …
    $ indoor_outdoor.biting.notes                           <chr> NA, NA, NA, NA, …
    $ biting.activity..indoor......No..of.sampling.nights   <dbl> NA, NA, NA, NA, …
    $ X18.00.19.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X19.00.20.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X20.00.21.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X21.00.22.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X22.00.23.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X23.00.00.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X00.00.01.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X01.00.02.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X02.00.03.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X03.00.04.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X04.00.05.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X05.00.06.00..in.                                     <dbl> NA, NA, NA, NA, …
    $ X18.30.21.30..in.                                     <lgl> NA, NA, NA, NA, …
    $ X21.30.00.30..in.                                     <lgl> NA, NA, NA, NA, …
    $ X00.30.03.30..in.                                     <lgl> NA, NA, NA, NA, …
    $ X03.30.06.30..in.                                     <lgl> NA, NA, NA, NA, …
    $ biting.activity..outdoor......No..of.sampling.nights  <dbl> NA, NA, NA, NA, …
    $ X18.00.19.00..out.                                    <lgl> NA, NA, NA, NA, …
    $ X19.00.20.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X20.00.21.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X21.00.22.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X22.00.23.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X23.00.00.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X00.00.01.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X01.00.02.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X02.00.03.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X03.00.04.00..out.                                    <dbl> NA, NA, NA, NA, …
    $ X04.00.05.00..out.                                    <lgl> NA, NA, NA, NA, …
    $ X05.00.06.00..out.                                    <lgl> NA, NA, NA, NA, …
    $ X18.30.21.30..out.                                    <lgl> NA, NA, NA, NA, …
    $ X21.30.00.30..out.                                    <lgl> NA, NA, NA, NA, …
    $ X00.30.03.30..out.                                    <lgl> NA, NA, NA, NA, …
    $ X03.30.06.30..out.                                    <lgl> NA, NA, NA, NA, …
    $ biting.activity..combined......No..of.sampling.nights <dbl> NA, NA, NA, NA, …
    $ X18.00.19.00..combined.                               <lgl> NA, NA, NA, NA, …
    $ X19.00.20.00..combined.                               <lgl> NA, NA, NA, NA, …
    $ X20.00.21.00..combined.                               <lgl> NA, NA, NA, NA, …
    $ X21.00.22.00..combined.                               <lgl> NA, NA, NA, NA, …
    $ X22.00.23.00..combined.                               <dbl> NA, NA, NA, NA, …
    $ X23.00.00.00..combined.                               <dbl> NA, NA, NA, NA, …
    $ X00.00.01.00..combined.                               <dbl> NA, NA, NA, NA, …
    $ X01.00.02.00..combined.                               <dbl> NA, NA, NA, NA, …
    $ X02.00.03.00..combined.                               <dbl> NA, NA, NA, NA, …
    $ X03.00.04.00..combined.                               <dbl> NA, NA, NA, NA, …
    $ X04.00.05.00..combined.                               <lgl> NA, NA, NA, NA, …
    $ X05.00.06.00..combined.                               <chr> NA, NA, NA, NA, …
    $ X18.30.21.30..combined.                               <lgl> NA, NA, NA, NA, …
    $ X21.30.00.30..combined.                               <dbl> NA, NA, NA, NA, …
    $ X00.30.03.30..combined.                               <lgl> NA, NA, NA, NA, …
    $ X03.30.06.30..combined.                               <lgl> NA, NA, NA, NA, …
    $ biting.notes                                          <chr> NA, NA, NA, NA, …
    $ resting.sampling..indoor.                             <chr> NA, NA, NA, NA, …
    $ unfed..indoor.                                        <dbl> NA, NA, NA, NA, …
    $ fed..indoor.                                          <dbl> NA, NA, NA, NA, …
    $ gravid..indoor.                                       <dbl> NA, NA, NA, NA, …
    $ total..indoor.                                        <dbl> NA, NA, NA, NA, …
    $ resting.sampling..outdoor.                            <chr> NA, NA, NA, NA, …
    $ unfed..outdoor.                                       <dbl> NA, NA, NA, NA, …
    $ fed..outdoor.                                         <dbl> NA, NA, NA, NA, …
    $ gravid..outdoor.                                      <dbl> NA, NA, NA, NA, …
    $ total..outdoor.                                       <dbl> NA, NA, NA, NA, …
    $ resting.sampling..other.                              <chr> NA, NA, NA, NA, …
    $ unfed..other.                                         <dbl> NA, NA, NA, NA, …
    $ fed..other.                                           <dbl> NA, NA, NA, NA, …
    $ gravid..other.                                        <dbl> NA, NA, NA, NA, …
    $ total..other.                                         <dbl> NA, NA, NA, NA, …
    $ resting..unit.                                        <chr> NA, NA, NA, NA, …
    $ resting.notes                                         <chr> NA, NA, NA, NA, …
    $ larval.habitat_1                                      <chr> NA, NA, NA, NA, …
    $ LH.1_site.character                                   <chr> NA, NA, NA, NA, …
    $ LH.1_turbidity                                        <chr> NA, NA, NA, NA, …
    $ LH.1_salinity                                         <chr> NA, NA, NA, NA, …
    $ LH.1_vegetation                                       <chr> NA, NA, NA, NA, …
    $ LH.1_shade                                            <chr> NA, NA, NA, NA, …
    $ LH.1_water.current                                    <chr> NA, NA, NA, NA, …
    $ LH.1_size..diameter_widest.point.                     <chr> NA, NA, NA, NA, …
    $ LH.1_depth                                            <chr> NA, NA, NA, NA, …
    $ LH.1_permanence                                       <chr> NA, NA, NA, NA, …
    $ LH.1_other.fauna                                      <chr> NA, NA, NA, NA, …
    $ LH.1_larval.control.present                           <chr> NA, NA, NA, NA, …
    $ larval.habitat_2                                      <chr> NA, NA, NA, NA, …
    $ LH.2_site.character                                   <chr> NA, NA, NA, NA, …
    $ LH.2_turbididity                                      <lgl> NA, NA, NA, NA, …
    $ LH.2_salinity                                         <lgl> NA, NA, NA, NA, …
    $ LH.2_vegetation                                       <chr> NA, NA, NA, NA, …
    $ LH.2_shade                                            <chr> NA, NA, NA, NA, …
    $ LH.2_water.current                                    <lgl> NA, NA, NA, NA, …
    $ LH.2_size                                             <chr> NA, NA, NA, NA, …
    $ LH.2_depth                                            <chr> NA, NA, NA, NA, …
    $ LH.2_permenance                                       <chr> NA, NA, NA, NA, …
    $ LH.2_other.fauna                                      <chr> NA, NA, NA, NA, …
    $ LH.2_larval.control.present                           <chr> NA, NA, NA, NA, …
    $ Larval.habitat_3                                      <chr> NA, NA, NA, NA, …
    $ LH.1_site.character_3                                 <chr> NA, NA, NA, NA, …
    $ LH.3_turbidity                                        <chr> NA, NA, NA, NA, …
    $ LH.3_salinity                                         <lgl> NA, NA, NA, NA, …
    $ LH.3_vegetation                                       <chr> NA, NA, NA, NA, …
    $ LH.3_shade                                            <lgl> NA, NA, NA, NA, …
    $ LH.3_water.current                                    <lgl> NA, NA, NA, NA, …
    $ LH.3_size                                             <chr> NA, NA, NA, NA, …
    $ LH.3_depth                                            <chr> NA, NA, NA, NA, …
    $ LH.3_other.fauna                                      <chr> NA, NA, NA, NA, …
    $ LH.3_larval.control.present                           <lgl> NA, NA, NA, NA, …
    $ larval.notes                                          <chr> NA, NA, NA, NA, …
    $ data.abstracted.by                                    <chr> NA, "LC", "LC", …
    $ data.checked.by                                       <chr> NA, "DNN", "DNN"…
    $ final.check                                           <chr> NA, NA, NA, NA, …
    $ anopheline_region_id                                  <lgl> NA, NA, NA, NA, …
    $ anopheline_taxonomic_level                            <lgl> NA, NA, NA, NA, …
    $ Larval.notes                                          <chr> NA, NA, NA, NA, …
    $ Larval.site.notes                                     <chr> NA, NA, NA, NA, …
    $ continent                                             <chr> "Africa", "Afric…

#### old data dump

read in, and get rid of non-African sites

``` r
old_db <- read_csv(
  file = "2024-03-14 - VA old but updated data for cleaning/OLD MAP database dump/20210726_vector_extraction (1).csv"
) |>
  mutate( # assign a continent based on the 3-letter country ID
    # some vector_site_country entries are blank, but the ID is not
    continent = countrycode(
      vector_site_country_id,
      "iso3c",
      "continent"
    )
  ) |>
  filter(continent == "Africa") |> # only keep African countries
  select(-continent) # remove this column from the database
```

    Rows: 198490 Columns: 70
    ── Column specification ────────────────────────────────────────────────────────
    Delimiter: ","
    chr (37): sample_period_assi, collection_method_name, collection_method_abbr...
    dbl (23): sample_period_id, sample_period_month_start, sample_period_year_st...
    lgl (10): anopheline_is_dvs, vector_site_bestguess_good, vector_site_bestgue...

    ℹ Use `spec()` to retrieve the full column specification for this data.
    ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
glimpse(old_db)
```

    Rows: 39,119
    Columns: 70
    $ sample_period_id                         <dbl> 2277, 2277, 2278, 2278, 2278,…
    $ sample_period_month_start                <dbl> 7, 7, 7, 7, 7, 7, 7, 7, 7, 7,…
    $ sample_period_year_start                 <dbl> 1994, 1994, 1994, 1994, 1994,…
    $ sample_period_month_end                  <dbl> 12, 12, 12, 12, 12, 12, 12, 1…
    $ sample_period_year_end                   <dbl> 1994, 1994, 1994, 1994, 1994,…
    $ sample_period_assi                       <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ sample_period_control_type_id            <dbl> NA, NA, NA, NA, NA, NA, NA, N…
    $ collection_id                            <dbl> 6525, 6525, 6526, 6527, 6528,…
    $ collection_ordinal                       <dbl> 1, 1, 1, 2, 3, 4, 1, 1, 2, 2,…
    $ collection_method_id                     <dbl> 9, 9, 12, 13, 6, 19, 12, 12, …
    $ collection_count                         <dbl> 616, 616, NA, NA, NA, NA, NA,…
    $ collection_method_name                   <chr> "larval collection", "larval …
    $ collection_method_abbreviation           <chr> "l", "l", "mbi", "mbo", "hri"…
    $ identification_ordinal                   <dbl> 1, 2, 1, 1, 1, 1, 1, 2, 1, 2,…
    $ identification_method_name               <chr> "morphology", "chromosomal ba…
    $ identification_method_abbreviation       <chr> "m", "cbs", "m", "m", "m", "m…
    $ control_method_name                      <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ control_method_abbreviation              <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ sample_period_sample_aggregate           <dbl> 616, 616, 10, 10, 10, 10, 116…
    $ sample_period_notes                      <chr> NA, NA, "Paper also gives mor…
    $ anopheline_id                            <dbl> 7, 7, 23, 23, 23, 23, 7, 7, 7…
    $ anopheline_name                          <chr> "<i>Anopheles arabiensis</i> …
    $ anopheline_sub_genus                     <chr> "Cellia", "Cellia", "Cellia",…
    $ anopheline_species                       <chr> "arabiensis", "arabiensis", "…
    $ anopheline_abbreviation                  <chr> "arabiensis", "arabiensis", "…
    $ anopheline_author                        <chr> "Patton, 1905", "Patton, 1905…
    $ anopheline_parent_id                     <dbl> 54, 54, NA, NA, NA, NA, 54, 5…
    $ anopheline_scientific_name               <chr> "<i>Anopheles</i> (<i>Cellia<…
    $ anopheline_taxonomic_level               <chr> "species", "species", "specie…
    $ anopheline_region_id                     <chr> "Africa", "Africa", "Africa",…
    $ anopheline_scientific_abbr               <chr> "<i>An. arabiensis</i>", "<i>…
    $ anopheline_is_dvs                        <lgl> TRUE, TRUE, TRUE, TRUE, TRUE,…
    $ anophenline_taxonomic_level              <chr> "species", "species", "specie…
    $ anopheline_reboot_id                     <dbl> NA, NA, NA, NA, NA, NA, NA, N…
    $ anopheline_reboot_parent                 <dbl> NA, NA, NA, NA, NA, NA, NA, N…
    $ anopheline_reboot_species                <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ anopheline_reboot_full_name              <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ anopheline_reboot_species_alternate      <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ anopheline_reboot_full_name_alternate    <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ anopheline_reboot_sub_genus              <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ vector_site_id                           <dbl> 28821, 28821, 28821, 28821, 2…
    $ vector_site_coordinates_name             <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ vector_site_coordinates_ordinal          <dbl> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…
    $ vector_site_coordinates_latitude         <dbl> 7.984, 7.984, 7.984, 7.984, 7…
    $ vector_site_coordinates_longitude        <dbl> 38.716, 38.716, 38.716, 38.71…
    $ vector_site_admin2_id                    <dbl> 13728, 13728, 13728, 13728, 1…
    $ vector_site_full_name                    <chr> "Eddo Kontola (Zwai Area)", "…
    $ vector_site_latlong_source               <chr> "Combination", "Combination",…
    $ vector_site_bestguess_good               <lgl> NA, NA, NA, NA, NA, NA, NA, N…
    $ vector_site_bestguess_rough              <lgl> NA, NA, NA, NA, NA, NA, NA, N…
    $ vector_site_admin1_paper                 <chr> NA, NA, NA, NA, NA, NA, NA, N…
    $ vector_site_admin2_paper                 <chr> "Eastern Shoa Zone", "Eastern…
    $ vector_site_admin3_paper                 <chr> "Adami Tulu District", "Adami…
    $ vector_site_vector_site_notes            <chr> "Rift Valley", "Rift Valley",…
    $ vector_site_notes                        <chr> "Descirption in report + Enca…
    $ vector_site_forest                       <lgl> FALSE, FALSE, FALSE, FALSE, F…
    $ vector_site_rice                         <lgl> FALSE, FALSE, FALSE, FALSE, F…
    $ vector_site_country                      <chr> "Ethiopia", "Ethiopia", "Ethi…
    $ vector_site_has_vector_data              <lgl> TRUE, TRUE, TRUE, TRUE, TRUE,…
    $ vector_site_has_geometry                 <lgl> TRUE, TRUE, TRUE, TRUE, TRUE,…
    $ vector_site_admin_id                     <dbl> 13763, 13763, 13763, 13763, 1…
    $ vector_site_country_id                   <chr> "ETH", "ETH", "ETH", "ETH", "…
    $ vector_site_area_type                    <chr> "point", "point", "point", "p…
    $ vector_site_rural_urban                  <chr> "rural", "rural", "rural", "r…
    $ source_id                                <dbl> 351, 351, 351, 351, 351, 351,…
    $ source_citation                          <chr> "Abose, T. (1998). <i>Re-orie…
    $ sample_period_complex                    <chr> "n.a.", "n.a.", "n.a.", "n.a.…
    $ sample_period_tag_recommended_unreliable <lgl> FALSE, FALSE, FALSE, FALSE, F…
    $ sample_period_is_copy                    <lgl> FALSE, FALSE, FALSE, FALSE, F…
    $ sample_period_asia_2016_dataset          <lgl> FALSE, FALSE, FALSE, FALSE, F…

Get site data from old database. This is unique combinations of site id,
site name, latitude, and longitude

``` r
old_sites <- old_db |>
  select(
    #id = vector_site_id,
    site = vector_site_full_name,
    lat = vector_site_coordinates_latitude,
    lon = vector_site_coordinates_longitude
  ) |>
  distinct()

old_sites
```

    # A tibble: 2,444 × 3
       site                       lat   lon
       <chr>                    <dbl> <dbl>
     1 Eddo Kontola (Zwai Area)  7.98  38.7
     2 Kwameta                  -5.11  38.5
     3 Magundi                  -5.09  38.5
     4 Kwamhanya                -5.06  38.5
     5 Bagamoyo                 -5.07  38.4
     6 Balangai                 -4.93  38.4
     7 Ebolakounou               3.93  12.1
     8 Koundou                   3.9   12.1
     9 Labe                     11.3  -12.3
    10 Sombili                  11.5  -12.5
    # ℹ 2,434 more rows

There are 2444 unique combinations. However, some site names correspond
to \>1 set of coordinates or vice versa, as there are fewer unique site
names than site IDs, and no site names are missing:

``` r
length(unique(old_sites$site)) # number of unique site names
```

    [1] 1904

``` r
sum(is.na(old_sites$site)) # number of missing site names
```

    [1] 0

Let’s find the duplicated names, and check for duplicated coordinates.
Here are all the sites with either duplicated names or duplicated
coordinates

``` r
old_dup <- old_sites |>
  group_by(site) |>
  mutate(
    unique_site = n(),
    unique_site = ifelse(unique_site == 1, TRUE, FALSE)
  ) |>
  ungroup () |>
  group_by(lon, lat) |>
  mutate(
    unique_coords = n(),
    unique_coords = ifelse(unique_coords == 1, TRUE, FALSE)
  ) |>
  filter(!unique_site | !unique_coords) |>
  arrange(site)

old_dup
```

    # A tibble: 1,070 × 5
    # Groups:   lon, lat [914]
       site                                    lat     lon unique_site unique_coords
       <chr>                                 <dbl>   <dbl> <lgl>       <lgl>        
     1 3 Rice Field Districts in Bouake Ci…  7.69   -5.03  TRUE        FALSE        
     2 3 villages (Diomandou Walo, Diomand… 16.5   -14.4   TRUE        FALSE        
     3 4 contiguous villages near Asembo    -0.183  34.4   TRUE        FALSE        
     4 Abia                                  5.72    0.133 FALSE       TRUE         
     5 Abia                                  5.72    0.132 FALSE       TRUE         
     6 Abia                                  5.72   -0.133 FALSE       TRUE         
     7 Abidjan                               5.33   -4.03  FALSE       TRUE         
     8 Abidjan                               5.33   -4.03  FALSE       TRUE         
     9 Abidjan                               5.34   -4.03  FALSE       TRUE         
    10 Abuda                                15.4    36.5   TRUE        FALSE        
    # ℹ 1,060 more rows

There are 1070 entries where the site name or coordinates or both are
duplicated.

We have four possible groups based on whether site and/or coordinates
are unique:

<table>
<colgroup>
<col style="width: 20%" />
<col style="width: 20%" />
<col style="width: 60%" />
</colgroup>
<thead>
<tr class="header">
<th>Unique site</th>
<th>Unique coordinates</th>
<th>Implication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>TRUE</td>
<td>TRUE</td>
<td>Data are good to go</td>
</tr>
<tr class="even">
<td>FALSE</td>
<td>TRUE</td>
<td>Multiple locations with single name. This is a problem ; not
matchable as is. Coods may have a typo.</td>
</tr>
<tr class="odd">
<td>TRUE</td>
<td>FALSE</td>
<td>Same location with multiple names. Possible name typos. This is not
a problem and should be matchable if typo is carried over.</td>
</tr>
<tr class="even">
<td>FALSE</td>
<td>FALSE</td>
<td>Multiple names for same location AND multiple locations for same
name</td>
</tr>
</tbody>
</table>

#### Duplicated site; unique coordinates:

``` r
old_dup |>
  filter(
    !unique_site & unique_coords
  )
```

    # A tibble: 793 × 5
    # Groups:   lon, lat [793]
       site         lat    lon unique_site unique_coords
       <chr>      <dbl>  <dbl> <lgl>       <lgl>        
     1 Abia        5.72  0.133 FALSE       TRUE         
     2 Abia        5.72  0.132 FALSE       TRUE         
     3 Abia        5.72 -0.133 FALSE       TRUE         
     4 Abidjan     5.33 -4.03  FALSE       TRUE         
     5 Abidjan     5.33 -4.03  FALSE       TRUE         
     6 Abidjan     5.34 -4.03  FALSE       TRUE         
     7 Abuja       9.18  7.17  FALSE       TRUE         
     8 Abuja       9.18  7.17  FALSE       TRUE         
     9 Adibosqual 15.7  38.6   FALSE       TRUE         
    10 Adibosqual 15.7  38.6   FALSE       TRUE         
    # ℹ 783 more rows

So 665 entries have site names associated with \>1 sets of coordinates.
Some look like typos in the coordinates (e.g., Abia), others look like
different nearby locations given the same name (Abuja)

``` r
old_dup |>
  filter(
    !unique_site & unique_coords
  ) |>
  pull(site) |>
  unique() |>
  length()
```

    [1] 326

Of which there are 326 unique site names.

#### Unique site; duplicated coordinates

``` r
old_dup |>
  filter(
    unique_site & !unique_coords
  ) |>
  arrange(lat, lon)
```

    # A tibble: 200 × 5
    # Groups:   lon, lat [111]
       site                     lat   lon unique_site unique_coords
       <chr>                  <dbl> <dbl> <lgl>       <lgl>        
     1 Nkunduze               -27.8  32.2 TRUE        FALSE        
     2 Nkunduzi               -27.8  32.2 TRUE        FALSE        
     3 Opansi                 -27.6  32.3 TRUE        FALSE        
     4 Mahlabaneni            -26.8  31.8 TRUE        FALSE        
     5 Mahlabaneni (Big Bend) -26.8  31.8 TRUE        FALSE        
     6 Mahlabaneni (big bend) -26.8  31.8 TRUE        FALSE        
     7 Luis  cabral           -25.9  32.5 TRUE        FALSE        
     8 Luis Cabral            -25.9  32.5 TRUE        FALSE        
     9 Hectorspruit           -25.4  31.9 TRUE        FALSE        
    10 Knp, SKUKUZA           -25.0  31.6 TRUE        FALSE        
    # ℹ 190 more rows

There are 200 of these which look like they include typos in site names.
NB: some must include sites where the name is not unique and location
not unique

``` r
old_dup |>
  filter(
    unique_site & !unique_coords
  ) |>
  arrange(lat, lon)  |>
  pull(site) |>
  unique() |>
  length()
```

    [1] 200

These make 200 unique names, as expected.

#### Duplicated site; duplicated coordinates

Arranged by site name:

``` r
old_dup |>
  filter(
    !unique_site & !unique_coords
  ) |>
  arrange(site)
```

    # A tibble: 77 × 5
    # Groups:   lon, lat [58]
       site                          lat   lon unique_site unique_coords
       <chr>                       <dbl> <dbl> <lgl>       <lgl>        
     1 Afan-Essokie                2.38  10    FALSE       FALSE        
     2 Agbalilame                  6.38   2.50 FALSE       FALSE        
     3 Agbalilame                  6.4    2.52 FALSE       FALSE        
     4 Ahero                      -0.183 34.9  FALSE       FALSE        
     5 Ambodifotatra & Lonkintsy -17.0   49.9  FALSE       FALSE        
     6 Asembo                     -0.182 34.4  FALSE       FALSE        
     7 Asembo                     -0.182 34.4  FALSE       FALSE        
     8 Asembo                     -0.183 34.4  FALSE       FALSE        
     9 Asembo Bay                 -0.183 34.4  FALSE       FALSE        
    10 Asembo Bay                 -0.182 34.4  FALSE       FALSE        
    # ℹ 67 more rows

And coordinates:

``` r
old_dup |>
  filter(
    !unique_site & !unique_coords
  ) |>
  arrange(lat, lon)
```

    # A tibble: 77 × 5
    # Groups:   lon, lat [58]
       site                        lat   lon unique_site unique_coords
       <chr>                     <dbl> <dbl> <lgl>       <lgl>        
     1 Ophansi                   -27.6  32.3 FALSE       FALSE        
     2 Komatipoort               -25.4  31.9 FALSE       FALSE        
     3 Knp, Shingwedzi           -23.1  31.4 FALSE       FALSE        
     4 Knp, Shingwedzi           -23.1  31.4 FALSE       FALSE        
     5 Malahlapanga              -22.9  31.0 FALSE       FALSE        
     6 Manarintsoa               -19    47.4 FALSE       FALSE        
     7 Mahazina                  -19.0  47.3 FALSE       FALSE        
     8 Ambodifotatra & Lonkintsy -17.0  49.9 FALSE       FALSE        
     9 Kanyemba                  -15.7  30.3 FALSE       FALSE        
    10 Camussamba                -12.8  15.7 FALSE       FALSE        
    # ℹ 67 more rows

There are 77 of these. Some include Multiple spellings of the name
multiple times AND slight variations of coordinates, e.g. Asembo /
Asembo Bay.

Some entries will match with entries in other categories, i.e. with
unique s have spelling variations that are unique OR coordinates that
are unique:

``` r
old_dup |>
  filter(site %in% c("Ophansi", "Opansi"))
```

    # A tibble: 3 × 5
    # Groups:   lon, lat [2]
      site      lat   lon unique_site unique_coords
      <chr>   <dbl> <dbl> <lgl>       <lgl>        
    1 Opansi  -27.6  32.3 TRUE        FALSE        
    2 Ophansi -27.6  32.3 FALSE       TRUE         
    3 Ophansi -27.6  32.3 FALSE       FALSE        

## Matching

Sites, sorted by name in the new database:

``` r
new_sites <- new_db |>
  select(site, lat = latitude_1, lon = longitude_1) |>
  distinct() |>
  arrange(site)
new_sites
```

    # A tibble: 4,110 × 3
       site                                  lat   lon
       <chr>                               <dbl> <dbl>
     1 (El) Thamaniat                     16.0    32.6
     2 10 contiguous villages near Asembo -0.183  34.4
     3 10 villages (50 km NW Kisumu)       0.16   34.4
     4 10 villages near Farafenni         13.6   -15.6
     5 11 contiguous villages near Asembo -0.183  34.4
     6 11 villages (50 km NW Kisumu)       0.16   34.4
     7 12 contiguous villages near Asembo -0.183  34.4
     8 13 contiguous villages near Asembo -0.183  34.4
     9 14 contiguous villages near Asembo -0.183  34.4
    10 15 contiguous villages near Asembo -0.183  34.4
    # ℹ 4,100 more rows

*(NB: in these data it is evident that there are multiple site names for
the same location also in this new database, and multiple locations for
the same name)*

We can prioritise (well, are only interested in) matching the sites with
missing coordinates

``` r
missing_sites <- new_sites |>
  filter(is.na(lat)) |>
  select(site)

missing_sites
```

    # A tibble: 713 × 1
       site                                                                         
       <chr>                                                                        
     1 16 Villages                                                                  
     2 16 villages(Kahoko, Rubirizi, Rwempisi , Kitwe, Nyakafura, Kabimbiri , Bugun…
     3 17 Unspecified villages North of Bo.                                         
     4 31 villages along Malindi, Kilifi, Kwale district coastline                  
     5 32 sites in Cameroon                                                         
     6 Abdel Daiem                                                                  
     7 Abeokuta                                                                     
     8 Abidha_Boi                                                                   
     9 Abidjan                                                                      
    10 Abobo (Tegne)                                                                
    # ℹ 703 more rows

Try matching to `old_sites` and check how many matches are made.

``` r
site_match <- missing_sites |>
  left_join(
    y = old_sites,
    by = "site"
  ) |>
  group_by(site) |>
  mutate(
    n = n()
  ) |>
  ungroup() |>
  mutate(
    nmatches = case_when(
      is.na(lat) ~ 0,
      TRUE ~ n
    )
  ) |>
  select(-n)

site_match
```

    # A tibble: 885 × 4
       site                                                       lat   lon nmatches
       <chr>                                                    <dbl> <dbl>    <dbl>
     1 16 Villages                                              NA    NA           0
     2 16 villages(Kahoko, Rubirizi, Rwempisi , Kitwe, Nyakafu… NA    NA           0
     3 17 Unspecified villages North of Bo.                     NA    NA           0
     4 31 villages along Malindi, Kilifi, Kwale district coast… NA    NA           0
     5 32 sites in Cameroon                                     NA    NA           0
     6 Abdel Daiem                                              14.5  33.4         1
     7 Abeokuta                                                  7.16  3.35        1
     8 Abidha_Boi                                               NA    NA           0
     9 Abidjan                                                   5.33 -4.03        3
    10 Abidjan                                                   5.33 -4.03        3
    # ℹ 875 more rows

Sites with no match:

``` r
site_match |>
  filter(nmatches == 0)
```

    # A tibble: 371 × 4
       site                                                       lat   lon nmatches
       <chr>                                                    <dbl> <dbl>    <dbl>
     1 16 Villages                                                 NA    NA        0
     2 16 villages(Kahoko, Rubirizi, Rwempisi , Kitwe, Nyakafu…    NA    NA        0
     3 17 Unspecified villages North of Bo.                        NA    NA        0
     4 31 villages along Malindi, Kilifi, Kwale district coast…    NA    NA        0
     5 32 sites in Cameroon                                        NA    NA        0
     6 Abidha_Boi                                                  NA    NA        0
     7 Abobo (Tegne)                                               NA    NA        0
     8 Abu Rakham, Wad Awad, Tenedba                               NA    NA        0
     9 Abu-Eshrin                                                  NA    NA        0
    10 Abukako                                                     NA    NA        0
    # ℹ 361 more rows

**371 sites with no matching site name in the old data base.**

Sites with multiple matches:

``` r
site_match |>
  filter(nmatches > 1) |>
  print() |>
  write_csv("multiple_matches_20240409.csv")
```

    # A tibble: 264 × 4
       site         lat    lon nmatches
       <chr>      <dbl>  <dbl>    <dbl>
     1 Abidjan    5.33  -4.03         3
     2 Abidjan    5.33  -4.03         3
     3 Abidjan    5.34  -4.03         3
     4 Aguel-Hoc 19.5    0.854        2
     5 Aguel-Hoc 19.3    0.833        2
     6 Ahero     -0.183 34.9          6
     7 Ahero     -0.182 34.9          6
     8 Ahero     -0.181 34.9          6
     9 Ahero     -0.183 34.9          6
    10 Ahero     -0.184 34.9          6
    # ℹ 254 more rows

264 entries

``` r
site_match |>
  filter(nmatches > 1) |>
  pull(site) |>
  unique() |>
  length()
```

    [1] 92

92 unique site names with multiple sets of coordinates attached.
